java/src/IceGridGUI/src/main/java/com/zeroc/IceGridGUI/Coordinator.java

```diff
@@ -1250,419 +1252,375 @@ enum TrustDecision { YesAlways, YesThisTime, No };
-                                                                                           JOptionPane.ERROR_MESSAGE);
-                                                         });
-                            break;
-                        }
-                        catch(java.lang.InterruptedException e)
-                        {
-                        }
-                        catch(java.lang.reflect.InvocationTargetException e)
+                        while(true)
                         {
-                            break;
+                            try
+                            {
+                                SwingUtilities.invokeAndWait(() ->
+                                                            {
+                                                                JOptionPane.showMessageDialog(parent, ex.toString(), "Error loading keystore",
+                                                                                            JOptionPane.ERROR_MESSAGE);
+                                                            });
+                                break;
+                            }
+                            catch(java.lang.InterruptedException e)
+                            {
+                            }
+                            catch(java.lang.reflect.InvocationTargetException e)
+                            {
+                                break;
+                            }
                         }
+                        throw certificateException;
                     }
-                    return false;
-                }
-
-                boolean validDate = true;
-                boolean trustedCA = false;
-                boolean validAlternateName = false;
 
-                //
-                // Check the certificate date is valid.
-                //
+                    TrustDecision decision = new AcceptInvalidCertDialog().show(serverCertificate);
 
-                java.util.Date now = new java.util.Date();
-                if(now.getTime() > cert.getNotAfter().getTime() || now.getTime() < cert.getNotBefore().getTime())
-                {
-                    validDate = false;
-                }
-
-                String remoteAddress = null;
-                for(com.zeroc.Ice.ConnectionInfo p = info.underlying; p != null; p = p.underlying)
-                {
-                    if(p instanceof com.zeroc.Ice.IPConnectionInfo)
+                    if(decision == TrustDecision.YesThisTime)
                     {
-                        remoteAddress = ((com.zeroc.Ice.IPConnectionInfo)p).remoteAddress;
-                        break;
+                        _transientCert = serverCertificate;
+                        System.out.println("Trust this time");
+                        return;
                     }
-                }
-
-                //
-                // Check server alternate names match the connection remote address
```

Comment:
Are we no longer doing this?

Reply:
Before when we get a certificate exception we check:
- validity date
- hostname

And we use this to show additional info in the accept invalid dialog warning. Now we jus present the certificate details and let the user to decided if he wants to accept the untrusted certificate.

With the TrustManager API you don't have the info to check the remote host, and that is why I remove this.
We can keep the message for valid dates, but given that we should the expiration date doesn't seems to add much value.